sdk-setup/src/mb2

```diff
@@ -2124,7 +2145,9 @@ run_deploy() {
             else
                 method="pkcon --plain --noninteractive install-local"
             fi
-            ! behind_qtcreator || run_package || return
+            if [[ -z $no_build ]]; then
+                ! behind_qtcreator || run_package || return
+            fi
```

Comment:
The `run_package` step is only implied when running behind Qt Creator. At the same time your change on QtC side disables that by adding `--deploy-only` to every `mb2 deploy` invocation. So unless I missed something the same effect can be achieved by simply dropping the implied `run_package` execution here.

Reply:
The `run_package` step will be ignored only from `sfdk` if I checked correctly. So `--deploy-only` also need for use insede build engene

Reply:
`! behind_qtcreator || run_package || return` is the same as `if behind_qtcreator; then run_package || return; fi`